### DISCUSSION TITLE
Adding Barrel Switching to a weapon that doesn't support it.

### DISCUSSION THREAD

**Discussion by HotRossBuns** [acemod/ACE3#9807] [Q&A] [answered]
Hey everyone! I'm Ross!

I have been running a milsim unit and we are playing on the Western Sahara map. We really like the MG that comes with said DLC and it suites our purposes quite well, except it gets quite hot & you can not swap the barrel on it like with most other MGs. 
I would like (and have been trying to for hours) to add the ACE3 swap barrel functionality to the 'Vektor SS-77 Compact'. Is there a good way to do so that might be more intuitive for a beginner?

https://ace3.acemod.org/wiki/framework/overheating-framework

I found the information about the overheating framework above ^

However Im afraid the instructions are a bit over my head, im really out of my comfort zone here. I'm not exactly sure which PBO I should be interacting with, since its a Creator DLC weapon. There are multiple class CfgWeapons in various PBOs, all of which I try interacting with, fail. 

Please let me know if there is any additional information I can provide. I really haven't made many steps, as im so unsure where to even begin.

_I really appreciate the help! Cheers!_

**Comment by Drofseh** [comment, marked as answer]
#9808
